src/storage/mod.rs

```diff
@@ -19,22 +31,128 @@ pub trait Engine {
 /// table (like column names, column types, storage engine, ...). It's `access`
 /// method locks the table globally and returns a storage engine to access
 /// the table data.
-pub struct Table;
+
+pub enum DatabaseError{
+    Io(Error),
+    Bin(EncodingError),
+}
+
+impl From<io::Error> for DatabaseError {
+    fn from(err: io::Error) -> DatabaseError {
+        DatabaseError::Io(err)
+    }
+}
+
+impl  From<EncodingError> for DatabaseError {
+    fn from(err: EncodingError) -> DatabaseError {
+        DatabaseError::Bin(err)
+    }
+}
+
+/// A Enum for File Modes
+///
+/// Can be used to define the save and load configuration of open_file
+#[derive(Debug)]
+enum FileMode{LoadDefault, SaveDefault,}
+
+/// A Enum for Datatypes (will be removed later)
+#[repr(u8)]
+#[derive(Clone,Copy,Debug,RustcDecodable, RustcEncodable)]
+enum DataType{ Integer = 1, Float = 2,}
+impl DataType{
+    pub fn value(&self) -> u8{
+       *self as u8
+    }
+}
+
+pub struct Database {
+    name: String,
+}
+
+impl Database {
+    pub fn new_database(database_name: &str) -> Database{
+        Database{name: database_name.to_string()}
+    }
+    pub fn create_database(&self) -> Result<(),DatabaseError>{
+        println!("trying to create dir!");
+        try!(create_dir(&self.name));
+        println!("created dir");
+        Ok(())
+    }
+}
+
+#[derive(Debug,RustcDecodable, RustcEncodable)]
+pub struct Table {
+    engine_id: u8,
+    version_nmbr: u8,
+    magic_nmbr: u8,
+    column_nmbr: u16,
+    columns: Vec<Column>,
+
+}
 
 impl Table {
-    pub fn load(database: &str, table: &str) -> Table {
+    pub fn load(database: &str, table: &str) -> Result<(), DatabaseError> {
         // TODO: Read the .tbl file from disk and parse it
-        Table
+        let mut file = try!(Self::open_file(database,table,FileMode::LoadDefault));
+
+        let data: Table = decode_from(&mut file, SizeLimit::Infinite).unwrap();
+        println!("{:?}", data);
+        Ok(())
+    }
+
+    pub fn create_new() -> Table {
+        let mut t: Vec<Column> = Vec::new();
+        t.push(Column::create_new());
+        Table { engine_id: 3 , version_nmbr:  1, magic_nmbr: 170, column_nmbr: 1, columns: t}
+    }
+
+    pub fn save(&self,database: &str, table: &str) -> Result<(), DatabaseError> {
+        //call for open file
+        let mut file = try!(Self::open_file(database,table,FileMode::SaveDefault));
+
+        try!(encode_into(&self,&mut file,SizeLimit::Infinite));
+
+        //debug message all okay
+        println!("I Wrote my File");
+        Ok(())
+    }
+
+    fn open_file(database: &str, table: &str, mode: FileMode) -> Result<File, DatabaseError> {
+        //create new file or open new one
+        let st = String::from(format!("{}/{}",database,table));
```

Comment:
`format!` already returns a `String`, so you should be able to remove the `String::from()`.